### PR TITLE
[Merged by Bors] - chore: move `Logic.Equiv.TransferInstance` to `Algebra.Equiv.TransferInstance`

### PR DESCRIPTION
**Pull request by jcommelin**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Equiv/Basic.lean [hunk 1]
```diff
@@ -38,7 +38,7 @@ In this file we continue the work on equivalences begun in `Mathlib/Logic/Equiv/
     `eb : β₁ ≃ β₂` using `Prod.map`.
 
   More definitions of this kind can be found in other files.
-  E.g., `Mathlib/Logic/Equiv/TransferInstance.lean` does it for many algebraic type classes like
+  E.g., `Mathlib.Algebra.Equiv.TransferInstance.lean` does it for many algebraic type classes like
```
Comment (Ruben-VandeVelde) [inline Mathlib/Logic/Equiv/Basic.lean:41]:
```suggestion
  E.g., `Mathlib.Algebra.Equiv.TransferInstance` does it for many algebraic type classes like
```
or
```suggestion
  E.g., `Mathlib/Algebra/Equiv/TransferInstance.lean` does it for many algebraic type classes like
```

File: Mathlib.lean [hunk 2]
```diff
@@ -3537,7 +3537,7 @@ import Mathlib.Logic.Equiv.Option
 import Mathlib.Logic.Equiv.Pairwise
 import Mathlib.Logic.Equiv.PartialEquiv
 import Mathlib.Logic.Equiv.Set
-import Mathlib.Logic.Equiv.TransferInstance
+import Mathlib.Algebra.Equiv.TransferInstance
```
Comment (Ruben-VandeVelde) [inline Mathlib.lean:3540]: More to the point, re-sort the imports here (and perhaps everywhere else)

**Review by Ruben-VandeVelde — APPROVED** [review]

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**